peps/pep-0774.rst

```diff
@@ -0,0 +1,255 @@
+
+This approach: 
+
+* Provides the best end-to-end experience for those looking to build CPython 
+  with the JIT
+* Lessens the barrier to entry for those looking to contribute to the JIT 
+* Ensures builds remain reproducible and consistent across platforms without 
+  relying on external infrastructure or download mechanisms 
+* Eliminates variability introduced by network conditions or potential
+  discrepancies between hosted files and the CPython repository state, and 
+* Subjects stencils to the same review processes we have for all other JIT-related
+  code
+
+However, this approach does result in a slight increase in overall
+repository size. Comparing repo growth on commits over the past 90 days, the 
+difference between the actual commits and the same commits with stencils added
+amounts to a difference of 0.03 MB per stencil file. This is a small increase in
+the context of the overall repository size, which has grown by 2.55 MB in the
+same time period. For six stencil files, this amounts to an upper bound of 0.18 MB.
+
+These stencils could become larger in the future with changes to register
+allocation, which would introduce 5-6 variants per instruction in each stencil
+file (5-6x larger). However, if we ended up going this route, there are
+additional modifications we could make to stencil files that could help
+counteract this size increase (e.g., stripping comments, minimizing the
+stencils). 
+
+Specification
+=============
+
+This specification outlines the proposed changes to remove the build-time
+dependency on LLVM and the contributor experience if this PEP is accepted.
+
+Repository changes
+------------------
+
+The CPython repository would now host the pre-compiled JIT stencils in a new
+subdirectory in ``Tools/jit`` called ``stencils/``. At present, the JIT is tested
+and built for six platforms, so to start, we’d check in six stencil files. In
+the future, we may check in additional stencil files if support for additional
+platforms is desired or relevant.
+
+.. code-block:: bash
+
+    cpython/
+        Tools/
+            jit/
+                stencils/
+                    aarch64-apple-darwin.h 
+                    aarch64-unknown-linux-gnu.h
+                    i686-pc-windows-msvc.h 
+                    x86_64-apple-darwin.h
+                    x86_64-pc-windows-msvc.h 
+                    x86_64-pc-linux-gnu.h
+
+Workflow
+--------
+
+To cover the workflow changes, this section can be divided into two parts -
+building CPython with the JIT enabled and working on the JIT’s implementation. 
+
+Building CPython with the JIT
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Precompiled JIT stencil files will be stored in the ``Tools/jit/stencils``
+directory, with each file name corresponding to its target triple as outlined
+above. At build time, we determine whether to use the checked in stencils or to
+generate a new stencil for the user’s platform. Specifically, for contributors
```

Comment:
Minor request, but please could you replace `’` with `'` throughout? They're rendered as typographic quotation marks in the HTML output regardless, and future edits from those without editors that automatically insert fancy quotation marks lead to stylistic drift!
```suggestion
generate a new stencil for the user's platform. Specifically, for contributors
```